lib/kitchen/verifier/pester.rb

```diff
@@ -152,6 +157,10 @@ def run_command_script
       end
 
       def really_wrap_shell_code(code)
+        # hypothesis: if OS not windows (can we detect or assume from transport)
+        # write the wrapped shell code to file with the pwsh(-preview) shebang
+        # and execute the file
+        # leave as is for windows (but double check we can use pwsh & pwsh-preview too)
```

Comment:
I also don't know for sure, but there is very likely a way to determine from the transport and/or the instance itself what kind of OS it is, yeah. 

For non-Windows if we plan to invoke the files directly via shebang we'll also need to add some code to set the executable bit of the file(s).